cargo-dist/src/tasks.rs

```diff
@@ -1349,9 +1376,32 @@ impl<'pkg_graph> DistGraphBuilder<'pkg_graph> {
                 };
 
                 let target_is_windows = target.contains("windows");
-                let platform_exe_ext = if target_is_windows { ".exe" } else { "" };
+                let platform_exe_ext;
+                let platform_lib_prefix;
+                if target_is_windows {
+                    platform_exe_ext = ".exe";
+                    platform_lib_prefix = "";
+                } else {
+                    platform_exe_ext = "";
+                    platform_lib_prefix = "lib";
+                };
 
-                let file_name = format!("{binary_name}{platform_exe_ext}");
+                let platform_lib_ext = if target_is_windows {
+                    ".dll"
+                } else if target.contains("linux") {
+                    ".so"
+                } else if target.contains("darwin") {
+                    ".dylib"
```

Comment:
bsds also use this, fwiw

Reply:
Oh is that where Apple got it from! That makes sense. We don't support BSDs atm, but worth keeping in mind for the future

Reply:
Thought: now that axoproject is in-tree, maybe in the future, we try to put all "what's the file extension/prefix" logic into axoproject's `platforms` mod.